wiki/Gameplay/Score/Total_score/id.md

```diff
@@ -0,0 +1,20 @@
+# Skor total
+
+*Untuk penggunaan istilah "skor" lainnya, lihat [Skor (disambiguasi)](/wiki/Disambiguation/Score).*
+
+**Skor total** merupakan total seluruh skor dari seluruh beatmap-beatmap yang pernah dimainkan oleh pemain, termasuk permainan-permainan yang gagal.
+
+Jika pemain memainkan sebuah beatmap ranked untuk pertama kalinya dan mendapatkan skor 500.000, skor total dan skor ranked-nya bertambah sebesar 500.000. Kemudian jika pemain memainkan ulang beatmap tersebut dan mendapatkan skor 400.000, maka hanya skor total pemain tersebut yang bertambah sebesar 400.000.
+
+## Level
+
+Level seorang pemain hanya bergantung pada **skor total** pemain tersebut. Skor yang dibutuhkan untuk menyelesaikan sebuah level didapatkan dari rumus berikut ini:
+
+```
+skor(n) = 5.000 / 3 * (4n^3 - 3n^2 - n) + 1,25 * 1,8^(n - 60) jika n <= 100
+skor(n) = 26.931.190.827 + 99.999.999.999 * (n - 100)         jika n > 100
+```
+
+Perlu diingat bahwa permainan ini melakukan pembulatan skor sehingga nilai untuk level 1 hingga 100 tidak seakurat rumus di atas. Hal ini disebabkan oleh permainan yang sudah membuat daftar selisih skor antar-level dan setiap nilai pada daftar tersebut dibulatkan. Sebagai contoh, untuk mencapai level 81 dari level 80 dibutuhkan skor sebanyak berikut ini: `skor(81) - skor(80) = 128.927.482,36216`, nilai tersebut dibulatkan menjadi 128.927.482. Saat pemain mencapai level 100, skor totalnya lebih besar 1,629 daripada hasil rumus yang seharusnya. Sehingga hasil rumus `skor(100) = 26.931.190.828,629` tidak cocok dengan rumus yang kedua.
```

Comment:
``Mengingat dalam perhitungannya osu! membulatkan nilai-nilai yang ada ke atas, dalam kasus-kasus tertentu perhitungan level 1 hingga 100 tidak akan mengikuti rumus di atas. Hal ini disebabkan oleh sistem osu! yang lebih mengutamakan perhitungan level berdasarkan selisih skor antar level sebelum nilai-nilai yang ada dibulatkan. Sebagai contoh, untuk dapat naik dari level 80 ke level 81, seorang pemain akan membutuhkan skor sebesar `skor(81) - skor(80) = 128.927.482,36216` yang dibulatkan oleh sistem menjadi 128.927.482. Pada saat pemain tersebut mencapai level 100, skor totalnya akan lebih besar 1.629 daripada skor total yang seharusnya, yang menyebabkan nilai `skor(100) = 26.931.190.828,629` menjadi tidak sesuai dengan hasil perhitungan rumus yang kedua.``

i genuinely have no idea what this paragraph means (even after reading the original english article lol) but the translation that i provided should be much more representative of what it is i suppose :x

Reply:
Something tells me there is something wrong with the numbers. I checked `Score(81)-Score(80)` is equals to `128,875,962.362164`. I'm using this function:

```cs
public static double CalculateScore(int i)
{
  double n = 0;
  n = 5000 / 3 * (4*Math.Pow(i,3) - 3*Math.Pow(i,2) - i) + 1.25 * Math.Pow(1.8, (i - 60));
  return n;
}
```

Reply:
your function is incorrect due to integer division of `5000 / 3`. it should be `5000.0 / 3`

Reply:
Ah I see, how about level 100? I calculated `score(100) = 26,931,190,828.6291` should be rounded to `26,931,190,829` not `26,931,190,827`?

Reply:
you've misunderstood the paragraph. it explains precisely why that is the case.

the +1.629 error at level 100 can be derived like this

```cs
double accumulatedError = 0;
for (int i = 0; i < 100; ++i)
    accumulatedError += (Score(i + 1) - Score(i)) - Math.Round(Score(i + 1) - Score(i));
Console.WriteLine(accumulatedError); // 1.6290969848632817
```

and the actual score required for level 100 can be derived like this

```cs
double actualRequiredScore = 0;
for (int i = 0; i < 100; ++i)
    actualRequiredScore += Math.Round(Score(i + 1) - Score(i));
Console.WriteLine(actualRequiredScore); // 26931190827

```

Reply:
So to get to certain level it comes from sum of rounded difference between score(i+1) - score(i) to nearest integer?

Reply:
Wait I miscalculated it, I took `level = score(i)`

Reply:
<details><summary>Here is the full table up to 100. I think this will make it super clear. After 100, the formula (second part) is exact.</summary>

| Lvl | Required score | What the formula gives you | Error |
| :-: | :-- | :-- | :-- |
| 1 | 0 | 1.0860059681165062E-15 | 4.826693191628916E-16 |
| 2 | 30000 | 30000 | 4.826693191628916E-16 |
| 3 | 130000 | 130000 | 4.826693191628916E-16 |
| 4 | 340000 | 340000 | 4.826693191628916E-16 |
| 5 | 700000 | 700000 | 4.826693191628916E-16 |
| 6 | 1250000 | 1250000 | 4.826693191628916E-16 |
| 7 | 2030000 | 2030000 | 4.826693191628916E-16 |
| 8 | 3080000 | 3080000 | 4.826693191628916E-16 |
| 9 | 4440000 | 4440000 | 4.826693191628916E-16 |
| 10 | 6150000 | 6150000 | 4.826693191628916E-16 |
| 11 | 8250000 | 8250000 | 4.826693191628916E-16 |
| 12 | 10780000 | 10780000 | 4.826693191628916E-16 |
| 13 | 13780000 | 13780000 | 4.826693191628916E-16 |
| 14 | 17290000 | 17290000 | 4.826693191628916E-16 |
| 15 | 21350000 | 21350000 | 4.826693191628916E-16 |
| 16 | 26000000 | 26000000 | 4.826693191628916E-16 |
| 17 | 31280000 | 31280000 | 4.826693191628916E-16 |
| 18 | 37230000 | 37230000 | 4.826693191628916E-16 |
| 19 | 43890000 | 43890000 | 4.826693191628916E-16 |
| 20 | 51300000 | 51300000 | 4.826693191628916E-16 |
| 21 | 59500000 | 59500000 | 4.826693191628916E-16 |
| 22 | 68530000 | 68530000 | 4.826693191628916E-16 |
| 23 | 78430000 | 78430000 | 4.826693191628916E-16 |
| 24 | 89240000 | 89240000 | 4.826693191628916E-16 |
| 25 | 101000000 | 101000000 | 4.826693191628916E-16 |
| 26 | 113750000 | 113750000 | 4.826693191628916E-16 |
| 27 | 127530000 | 127530000 | 4.826693191628916E-16 |
| 28 | 142380000 | 142380000 | 4.826693191628916E-16 |
| 29 | 158340000 | 158340000.00000003 | 2.980232287036463E-08 |
| 30 | 175450000 | 175450000.00000003 | 2.980232287036463E-08 |
| 31 | 193750000 | 193750000.00000006 | 5.960464525805994E-08 |
| 32 | 213280000 | 213280000.0000001 | 8.940696764575525E-08 |
| 33 | 234080000 | 234080000.00000015 | 1.4901161242114587E-07 |
| 34 | 256190000 | 256190000.0000003 | 2.9802322435962246E-07 |
| 35 | 279650000 | 279650000.00000054 | 5.36441803461185E-07 |
| 36 | 304500000 | 304500000.00000095 | 9.536743168889194E-07 |
| 37 | 330780000 | 330780000.00000167 | 1.668930054193607E-06 |
| 38 | 358530000 | 358530000.00000304 | 3.0398368840275913E-06 |
| 39 | 387790000 | 387790000.0000054 | 5.424022675043217E-06 |
| 40 | 418600000 | 418600000.00000983 | 9.834766388422123E-06 |
| 41 | 451000000 | 451000000.00001764 | 1.7642974853998295E-05 |
| 42 | 485030000 | 485030000.00003177 | 3.1769275665765876E-05 |
| 43 | 520730000 | 520730000.00005716 | 5.716085434008228E-05 |
| 44 | 558140000 | 558140000.0001029 | 0.00010287761688280689 |
| 45 | 597300000 | 597300000.0001853 | 0.00018525123596239675 |
| 46 | 638250000 | 638250000.0003335 | 0.0003335475921635686 |
| 47 | 681030000 | 681030000.0006003 | 0.0006003379821782171 |
| 48 | 725680000 | 725680000.0010805 | 0.001080513000488764 |
| 49 | 772240000 | 772240000.001945 | 0.0019450187683110296 |
| 50 | 820750000 | 820750000.0035009 | 0.0035009384155278264 |
| 51 | 871250000 | 871250000.0063016 | 0.006301641464233881 |
| 52 | 923780000 | 923780000.011343 | 0.01134300231933642 |
| 53 | 978380000 | 978380000.0204175 | 0.02041745185852099 |
| 54 | 1035090000 | 1035090000.0367515 | 0.03675150871276904 |
| 55 | 1093950000 | 1093950000.0661526 | 0.06615257263183642 |
| 56 | 1155000000 | 1155000000.1190748 | 0.11907482147216845 |
| 57 | 1218280000 | 1218280000.2143347 | 0.21433472633361866 |
| 58 | 1283830000 | 1283830000.3858025 | 0.3858025074005132 |
| 59 | 1351690000 | 1351690000.6944444 | 0.6944444179534917 |
| 60 | 1421900001 | 1421900001.25 | 0.25000000000000044 |
| 61 | 1494500002 | 1494500002.25 | 0.25000000000000044 |
| 62 | 1569530004 | 1569530004.05 | 0.049999952316284624 |
| 63 | 1647030007 | 1647030007.29 | 0.2899999618530278 |
| 64 | 1727040013 | 1727040013.122 | 0.12199997901916548 |
| 65 | 1809600023 | 1809600023.6196 | 0.6196000576019292 |
| 66 | 1894750042 | 1894750042.51528 | 0.5152800083160405 |
| 67 | 1982530076 | 1982530076.527504 | 0.5275039672851567 |
| 68 | 2072980137 | 2072980137.7495072 | 0.7495071887969975 |
| 69 | 2166140247 | 2166140247.949113 | 0.9491128921508794 |
| 70 | 2262050445 | 2262050446.3084035 | 1.3084034919738774 |
| 71 | 2360750802 | 2360750803.355126 | 1.3551259040832524 |
| 72 | 2462281445 | 2462281446.039227 | 1.0392270088195805 |
| 73 | 2566682602 | 2566682602.8706083 | 0.8706083297729497 |
| 74 | 2673994684 | 2673994685.1670947 | 1.1670947074890141 |
| 75 | 2784258432 | 2784258433.3007708 | 1.30077075958252 |
| 76 | 2897515179 | 2897515179.941387 | 0.9413871765136723 |
| 77 | 3013807323 | 3013807323.894497 | 0.8944969177246098 |
| 78 | 3133179182 | 3133179183.010094 | 1.0100941658020024 |
| 79 | 3255678528 | 3255678529.4181695 | 1.418169498443604 |
| 80 | 3381359352 | 3381359352.952705 | 0.9527049064636235 |
| 81 | 3510286834 | 3510286835.314869 | 1.3148689270019536 |
| 82 | 3642546302 | 3642546303.566764 | 1.5667638778686528 |
| 83 | 3778259345 | 3778259346.420175 | 1.4201750755310063 |
| 84 | 3917612822 | 3917612823.5563154 | 1.556315422058106 |
| 85 | 4060911081 | 4060911082.4013677 | 1.4013676643371586 |
| 86 | 4208669947 | 4208669948.322462 | 1.3224620819091801 |
| 87 | 4361785906 | 4361785906.980432 | 0.9804315567016606 |
| 88 | 4521840632 | 4521840632.564777 | 0.5647773742675786 |
| 89 | 4691649138 | 4691649138.6166 | 0.6166000366210942 |
| 90 | 4876246449 | 4876246449.509879 | 0.5098791122436528 |
| 91 | 5084663609 | 5084663609.117783 | 0.11778259277343794 |
| 92 | 5333124496 | 5333124496.412008 | 0.4120082855224614 |
| 93 | 5650800093 | 5650800093.5416155 | 0.54161548614502 |
| 94 | 6090166168 | 6090166168.374908 | 0.37490844726562544 |
| 95 | 6745647103 | 6745647103.074835 | 0.07483482360839888 |
| 96 | 7787174785 | 7787174785.534702 | 0.5347023010253911 |
| 97 | 9520594613 | 9520594613.962463 | 0.9624633789062504 |
| 98 | 12496396304 | 12496396305.132437 | 1.1324367523193364 |
| 99 | 17705429348 | 17705429349.238388 | 1.238388061523438 |
| 100 | 26931190827 | 26931190828.629097 | 1.6290969848632817 |
</details>

Reply:
> Mengingat dalam perhitungannya osu! membulatkan nilai-nilai yang ada ke angka bulat terdekat, dalam kasus-kasus tertentu perhitungan level 1 hingga 100 tidak akan sepenuhnya sesuai dengan rumus di atas. Hal ini disebabkan oleh sistem osu! yang lebih mengutamakan perhitungan level berdasarkan selisih skor antar level sebelum nilai-nilai yang ada dibulatkan. Sebagai contoh, untuk dapat naik dari level 80 ke level 81, seorang pemain akan membutuhkan skor sebesar `skor(81) - skor(80) = 128.927.482,36216` yang dibulatkan oleh sistem menjadi 128.927.482. Pada saat pemain tersebut mencapai level 100, skor totalnya akan lebih besar 1,629 daripada skor total yang seharusnya, yang menyebabkan nilai `skor(100) = 26.931.190.828,629` menjadi tidak sesuai dengan hasil perhitungan rumus yang kedua.

This should be the final form of paragraph. I change the first sentence to match "round to the nearest integer" and 1.629 to 1,629. I'm glad it's quite clear now.